lib/auth/trustedcluster.go

```diff
@@ -520,24 +521,54 @@ func (a *Server) validateTrustedCluster(ctx context.Context, validateRequest *Va
 	}
 
 	// export local cluster certificate authority and return it to the cluster
-	validateResponse := ValidateTrustedClusterResponse{
-		CAs: []types.CertAuthority{},
+	validateResponse := ValidateTrustedClusterResponse{}
+
+	validateResponse.CAs, err = validationTrustedClusterCerts(ctx, a, domainName, validateRequest)
+	if err != nil {
+		return nil, trace.Wrap(err)
 	}
-	for _, caType := range []types.CertAuthType{types.HostCA, types.UserCA, types.DatabaseCA} {
-		certAuthority, err := a.GetCertAuthority(
+
+	// log the local certificate authorities we are sending
+	log.Debugf("Sending validate response: CAs=%v", validateResponse.CAs)
+
+	return &validateResponse, nil
+}
+
+func validationTrustedClusterCerts(ctx context.Context, srv *Server, domainName string, validateRequest *ValidateTrustedClusterRequest) ([]types.CertAuthority, error) {
```

Comment:
I would refactor this function a bit and made it just return appropriate CA types, e.g. `getCATypesForLeaf`. It would be a bit easier to reason about it and extend as right now it does 2 things: determines which CAs to send and fetches them from the backend.